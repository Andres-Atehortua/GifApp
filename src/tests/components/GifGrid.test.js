import React from "react";
import GifGrid from "components/GifGrid";
import { shallow } from "enzyme";
import useFetchGifs from "hooks/useFetchGifs";
// Hacemos un seguimiento de la funcion
jest.mock("hooks/useFetchGifs");

describe("Pruebas para el componente <GifGrid />", () => {
  const category = "Goku";

  test("should match snapshot", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);

    const h3 = wrapper.find("h3").text();
    expect(h3).toBe(category);

    expect(wrapper).toMatchSnapshot();
  });

  test("should show items when images are fetched useFetchGifs", () => {
    const gifs = [
      { id: 123, url: "https://localhost/cosa.jpg", title: "Titulo" },
    ];
    // Fingimos que la funcion nos regresa data
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    // Comprobar si un elemento existe
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
