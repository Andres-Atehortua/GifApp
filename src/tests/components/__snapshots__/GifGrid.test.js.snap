// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para el componente <GifGrid /> should match snapshot 1`] = `
<Fragment>
  <h3
    className=" animate__animated animate__zoomInDown"
  >
    Goku
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
